ddtrace/contrib/botocore/patch.py

```diff
@@ -52,23 +53,87 @@ def inject_trace_data_to_message_attributes(trace_data, entry):
         log.debug("skipping trace injection, max number (10) of MessageAttributes exceeded")
 
 
-def inject_trace_to_sqs_batch_message(args, span):
+def inject_trace_to_sqs_or_sns_batch_message(args, span):
     trace_data = {}
     HTTPPropagator.inject(span.context, trace_data)
     params = args[1]
 
-    for entry in params["Entries"]:
+    entries = params.get("Entries", params.get("PublishBatchRequestEntries", []))
+    for entry in entries:
         inject_trace_data_to_message_attributes(trace_data, entry)
 
 
-def inject_trace_to_sqs_message(args, span):
+def inject_trace_to_sqs_or_sns_message(args, span):
     trace_data = {}
     HTTPPropagator.inject(span.context, trace_data)
     params = args[1]
 
     inject_trace_data_to_message_attributes(trace_data, params)
 
 
+def inject_trace_to_event_bridge_detail(args, span):
+    params = args[1]
+    if "Entries" in params:
+        for entry in params["Entries"]:
+            if "Detail" in entry:
+                try:
+                    detail = json.loads(entry["Detail"])
+                    dd_context = detail.get("_datadog", {})
+                    HTTPPropagator.inject(span.context, dd_context)
+                    detail["_datadog"] = dd_context
+                    entry["Detail"] = json.dumps(detail)
```

Comment:
So you grab the "Detail" which i'm assuming is how you know it is an event bridge detail. Is this "Detail" that you are grabbing from entry populated by us or is that payload coming from AWS? (If its coming from AWS then how do you know other entries wont have "Detail" in it?)

Next you grab the existing dd_context if there is one, then you inject some HTTP context (not sure about that part), then you update the dd_context and finally update the "Detail" json with the new detail.
Does that sum up lines 80-84?

Reply:
Thanks for pointing this out, it seems like we might have missed something here. The `Detail` is there in these cases:

- a CloudWatch event (`Detail` generated by AWS)
- an EventBridge event (`aws.events` - `Detail` generated by AWS)
- an EventBridge event (`custom event` - `Detail` generated by us)

We need to add additional conditions to the make sure we only do the injection for the third case mentioned above.